File: java-starter/hello-service/src/main/java/org/example/HelloApp.java
package org.example;

import io.nitric.Nitric;

public class HelloApp {
    public static void main(String[] args) {
        var app = Apis.publicApi();

        app.get("/goodbye/:name", (ctx) -> {
            var name = ctx.getReq().getParams().getOrDefault("name", "world");

            ctx.getResp().text(String.format("Goodbye %s!", name));

            return ctx;
        });

        Nitric.INSTANCE.run();
    }
}

File: kotlin-starter/hello-service/src/main/kotlin/HelloApp.kt
import io.nitric.Nitric
import io.nitric.common.PublicApi

fun main() {
    val api = PublicApi()

    api.get("/hello/:name") { ctx ->
        val name = ctx.req.params["name"]

        ctx.resp.text("hello $name")

        ctx
    }

    Nitric.run()
}